### PR TITLE
Create API endpoint for retrieving logs

### PR DESCRIPTION
**Pull request by rudigiesler**
Now that we store logs, and have created a reader to read the last `n` log entries, we should create an API endpoint that uses this to return the last `n` logs.

### CODE REVIEW

**Comment by rudigiesler** [comment]
Ready for review.

File: docs/http_api.rst [hunk 1]
```diff
@@ -165,6 +165,66 @@ Channels
...
 
+Logs
+^^^^
+
+.. http:get:: /channels/(channel_id:str)/logs
+
+   Get the most recent logs for a specific channel.
+
+   :query int n:
+       The amount of logs to fetch.
```
Comment (hodgestar) [inline docs/http_api.rst]: s/amount/number/ -- maybe a bikeshed? I think "amount" is for continuous quantities. We need a grammar expert. :)

File: docs/http_api.rst [hunk 2]
```diff
@@ -165,6 +165,66 @@ Channels
...
+   :param str logger: The logger that created the log, usually the channel id.
+   :param int level:
+       The level of the logs. Corresponds to the levels found in the python
+       module :py:mod:`logging`.
+   :param float timestamp: Timestamp, in the format of seconds since the epoch.
+   :param str message: The message of the log.
+
+   In the case of an exception, the following fields may also be returned:
+
+   :param str class: The class of the exception.
```
Comment (hodgestar) [inline docs/http_api.rst]: I wonder if we should nest these inside an `exception` sub-dict? Just thinking that `class` and `instance` by themselves don't really indicate that they are related to an exception?

File: docs/http_api.rst (hunk 1 above)
Comment (rudigiesler) [inline docs/http_api.rst]: Number sounds better :)

File: docs/http_api.rst (hunk 2 above)
Comment (rudigiesler) [inline docs/http_api.rst]: Sounds good, I'll make the change.

File: docs/http_api.rst [hunk 3]
```diff
@@ -165,6 +165,66 @@ Channels
...
+                level: 20,
+                timestamp: 987654321.0,
+                message: "Last log for the channel"
+            },
+            {
+                logger: "123-456-7a90",
+                level: 20,
+                timestamp: 987654320.0,
+                message: "Second last log for the channel"
+            }
```
Comment (hodgestar) [inline docs/http_api.rst]: Should we include an example of an exception log?

Comment (hodgestar) [inline docs/http_api.rst]: Should we include an example of an exception log?

Comment (rudigiesler) [inline docs/http_api.rst]: I added an example in https://github.com/praekelt/junebug/commit/d3cf551eae900f4c6036a50e5330a594d2617606

**Comment by hodgestar** [comment]
I think we need `get_logs` and API tests for the `n=None` case?

**Comment by hodgestar** [comment]
Other than the tests for `n=None`, looks good!

**Comment by rudigiesler** [comment]
Changes made. Ready for another review.

**Comment by hodgestar** [comment]
:+1:
